wgpu-core/src/device/mod.rs

```diff
@@ -515,7 +515,10 @@ impl<B: GfxBackend> Device<B> {
         let usage = conv::map_texture_usage(desc.usage, aspects);
 
         let mip_level_count = desc.mip_level_count;
-        if mip_level_count == 0 && mip_level_count > kind.compute_num_levels() as u32 {
+        if mip_level_count == 0
+            || mip_level_count > MAX_MIP_LEVELS
```

Comment:
This should be handled differently in the end: we need the max texture size in the limits, and we'd be validating that texture size here, and also making sure that `MAX_MIP_LEVELS` is sane at the adapter creation. This can be followed up with, not a blocker